Add a new page for deprecations, removals, and changes per version (#2053)
neo4j/docs-operations: modules/ROOT/pages/changes-deprecations-removals.adoc

|
| {check-mark}
| Replaced by xref:procedures.adoc#procedure_dbms_unquarantineDatabase[`dbms.unquarantineDatabase()`]
|===
====

=== Java API

In 2025.01, the deprecated public API is removed.

=== Neo4j configuration settings with changed default values

[options=header, cols="3m,1,1"]
|===
| Name
| Old default value
| New default value

|`dbms.cypher.infer_schema_parts`
|`OFF`
|`MOST_SELECTIVE_LABEL`

|`db.logs.query.annotation_data_format`
|`CYPHER`
|`JSON`

|`dbms.databases.seed_from_uri_providers`
|`S3SeedProvider,CloudSeedProvider`
|`CloudSeedProvider`

|`server.metrics.csv.rotation.compression`
|`NONE`
|`ZIP`

|`server.panic.shutdown_on_panic`
|`false`
|`true`

|===

=== Changed valid values of configuration settings

The `server.logs.config` is changed to depend on the provided path for the _conf_ directory instead of having the _conf_ directory encoded in the default values.

[options=header, cols="2m,3,3"]
|===
| Name
| Old valid value
| New valid value

|`server.logs.config`
|A path. If relative, it is resolved from `server.directories.neo4j_home`.
|A path. If relative, it is resolved from `server.directories.configuration`.
|===

=== Renamed configuration settings

> `dbms.cluster.discovery.endpoints` => `dbms.cluster.endpoints`

Author: ```suggestion
`dbms.cluster.discovery.v2.endpoints` => `dbms.cluster.endpoints`
```

> `dbms.kubernetes.service_port_name` => `dbms.kubernetes.discovery.service_port_name`

Author: ```suggestion
`dbms.kubernetes.v2.service_port_name` => `dbms.kubernetes.discovery.service_port_name`
```

:description: Page contains lists of procedures, configuration settings, metrics removed or deprecated in Neo4j 2025. Also, you can find information on changed defaults and new functionality of neo4j-admin commands.

//Check Mark
:check-mark: icon:check[]

[[removals-deprecations-2025-01]]
= Deprecations, removals, and changes per Neo4j version

> == Neo4j 2025.01

Reviewer: I thought this would be a page and not a section. Each page will have three headings: Changes, Deprecations, and Removals. In each section, we can list whatever should be listed either as a table or a paragraph if it is not a list. Deprecations should be ordered by month or minor version as well.

Author: I'm totally confused. Last time, you talked about all Neo4j versions on one page, not only 2025.01 but also 5.x series. Like it's done in the Cypher manual: all versions are on one page.
So, later we can add info about the 5.x series (retrospectively), and this page will be the same in 5.x and 2025 versions of the Operations manual.

I don't fully understand: what do you mean by: 'Each page will have three headings: Changes, Deprecations, and Removals... Deprecations should be ordered by month or minor version as well.'

Author: Could you please provide your idea on how to structure this section?

Reviewer: I am confused now as well. We need to take notes during our meetings. In my head, we agreed that we'd have a page for 2025 and 5 because there are a lot of versions in each of these major or year versions. And you said that you'll create the page for 2025. So, I thought the structure would be something like the following:

# Changes, deprecations, removals - this could be just a bullet on the left navigation
## 2025 - this is the first page
### 2025.01 release
#### Changes 
- default values
- whatever has been changed
#### Deprecations
- Procedures
- Configs
- Metrics
- whatever else has been deprecated
#### Removals
- Procedures
- Configs
- Metrics
- etc.

|label:enterprise[Enterprise Edition]

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_db.logs.query.annotation_data_as_json_enabled[`db.logs.query.annotation_data_as_json_enabled`]
|label:dynamic[]

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_db.tx_state.memory_allocation[`db.tx_state.memory_allocation`]
|

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_dbms.cluster.catchup.client_inactivity_timeout[`dbms.cluster.catchup.client_inactivity_timeout`]
|label:enterprise[Enterprise Edition]

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_dbms.cluster.discovery.log_level[`dbms.cluster.discovery.log_level`]
|label:enterprise[Enterprise Edition]

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_dbms.cluster.discovery.type[`dbms.cluster.discovery.type`]
|label:enterprise[Enterprise Edition]

> |link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_dbms.cluster.discovery.v2.endpoints[`dbms.cluster.discovery.v2.endpoints`]

Author: not removed, but renamed? cc @nick-giles-neo

Reviewer: This one is deprecated.

|<prefix>.causal_clustering.core.raft_log_entry_prefetch_buffer.size

|<prefix>.causal_clustering.core.raft_log_entry_prefetch_buffer.async_put

|<prefix>.causal_clustering.core.raft_log_entry_prefetch_buffer.sync_put

|<prefix>.causal_clustering.core.message_processing_delay

|<prefix>.causal_clustering.core.message_processing_timer

|<prefix>.causal_clustering.core.replication_new

|<prefix>.causal_clustering.core.replication_attempt

|<prefix>.causal_clustering.core.replication_fail

|<prefix>.causal_clustering.core.replication_maybe

|<prefix>.causal_clustering.core.replication_success

|<prefix>.causal_clustering.core.last_leader_message

a|**link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#read-replica-metrics[Read Replica metrics] - replaced accordingly by the link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#store-copy-metrics[Store copy metrics]**

|<prefix>.causal_clustering.read_replica.pull_updates

|<prefix>.causal_clustering.read_replica.pull_update_highest_tx_id_requested

|<prefix>.causal_clustering.read_replica.pull_update_highest_tx_id_received

|===
====

> === SSL framework

Author: I'm not sure about this section. We can rename it or hide it in meantime

Reviewer: I think it's fine.

> .See the CBC-based ciphers that are no longer used from Neo4j 2025.01:

Author: based on D.Pond's comments

* TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
* TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
* TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
* TLS_DHE_RSA_WITH_AES_128_CBC_SHA256

For more details, refer to the page on link:{neo4j-docs-base-uri}/operations-manual/5/security/ssl-framework/#ssl-other-configs[SSL framework] under the Security section.

> .Private keys

Author: Took it from the page, but I didn't ask anyone if it's true

Reviewer 2: @NataliaIvakina it hasn't been changed yet, so not yet removed in the product.  Was a Q1 item that was pushed out.

You can identify them by the line `-----BEGIN RSA PRIVATE KEY-----` at the beginning of the file.
While Neo4j 5.0 can load and use those keys, they are considered deprecated and will be removed in a future version.

.SSL-related configuration setting (or move it to the section 'Changed default values'?)

Starting from 2025.01, the default value of `dbms.ssl.policy.*.verify_hostname` is changed from `false` to `true`.
See details in the table link:{neo4j-docs-base-uri}/operations-manual/5/security/ssl-framework/#ssl-configuration[in the table under Configuration section on the page SSL framework].

=== Ports

Port `5000` is no longer used from 2025.01 onwards.

=== Deprecated built-in procedures

.See all deprecated procedures
[%collapsible]
====
[options=header, cols="3m,1,1,3"]
|===
| Name
| Community Edition
| Enterprise Edition
| Comment

| link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_quarantineDatabase[`dbms.quarantineDatabase()`] label:admin-only[]
|
| {check-mark}
| Replaced by xref:procedures.adoc#procedure_dbms_unquarantineDatabase[`dbms.unquarantineDatabase()`]
|===
====

> === Java API

Author: I don't know where to take information. I believe someone from Kernel may help

Reviewer: Misha can help with the Java API.

In 2025.01, the deprecated public API is removed.

> 

Author: ```suggestion

The `SeedProvider` is removed and replaced by the `DatabaseSeedProvider`.
For details, refer to the link:{neo4j-docs-base-uri}/java-reference/current/extending-neo4j/project-setup/#_databaseseedprovider[Java Reference -> Extending Neo4j -> Setting up a plugin project].
```

=== Neo4j configuration settings with changed default values

[options=header, cols="3m,1,1"]
|===
| Name
| Old default value
| New default value

|`dbms.cypher.infer_schema_parts`
|`OFF`
|`MOST_SELECTIVE_LABEL`

|`db.logs.query.annotation_data_format`
|`CYPHER`
|`JSON`

|`dbms.databases.seed_from_uri_providers`
|`S3SeedProvider,CloudSeedProvider`
|`CloudSeedProvider`

|`server.metrics.csv.rotation.compression`
|`NONE`
|`ZIP`

|`server.panic.shutdown_on_panic`
|`false`
|`true`

|===

=== Changed valid values of configuration settings

The `server.logs.config` is changed to depend on the provided path for the _conf_ directory instead of having the _conf_ directory encoded in the default values.

[options=header, cols="2m,3,3"]
|===
| Name
| Old valid value
| New valid value

|`server.logs.config`
|A path. If relative, it is resolved from `server.directories.neo4j_home`.
|A path. If relative, it is resolved from `server.directories.configuration`.
|===

=== Renamed configuration settings

> `dbms.cluster.discovery.endpoints` => `dbms.cluster.endpoints`

Author: @nick-giles-neo, is this setting is renamed or the `dbms.cluster.discovery.v2.endpoints`?

|The total number of Raft replication attempts that have failed. (counter)
|<prefix>.causal_clustering.core.replication_maybe
|Raft Replication maybe count. (counter)
|<prefix>.causal_clustering.core.replication_success
|The total number of Raft replication requests that have succeeded. (counter)
|<prefix>.causal_clustering.core.last_leader_message
|The time elapsed since the last message from a leader in milliseconds. Should reset periodically. (gauge)

2+|**link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#read-replica-metrics[Read Replica metrics] - replaced accordingly by the link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#store-copy-metrics[Store copy metrics]**

|<prefix>.causal_clustering.read_replica.pull_updates
|The total number of pull requests made by this instance. (counter)
|<prefix>.causal_clustering.read_replica.pull_update_highest_tx_id_requested
|The highest transaction id requested in a pull update by this instance. (counter)
|<prefix>.causal_clustering.read_replica.pull_update_highest_tx_id_received
|The highest transaction id that has been pulled in the last pull updates by this instance. (counter)

> 2+|**link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#discovery-service-V1[Discovery metrics v1] - removed without replacement. See xref:monitoring/metrics/reference/#discovery-service-metrics[Discovery metrics]**

Author: ```suggestion
2+|**link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#discovery-service-V1[Discovery metrics v1] - removed without replacement. See xref:monitoring/metrics/reference.adoc#discovery-service-metrics[Discovery metrics]**
```

| Enterprise Edition
| Comment

| link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_quarantineDatabase[`dbms.quarantineDatabase()`] label:admin-only[]
|
| {check-mark}
| Replaced by xref:procedures.adoc#procedure_dbms_unquarantineDatabase[`dbms.unquarantineDatabase()`]
|===
====

=== Java API

In 2025.01, the deprecated public API is removed.

The `SeedProvider` is removed and replaced by the `DatabaseSeedProvider`.
For details, refer to the link:{neo4j-docs-base-uri}/java-reference/current/extending-neo4j/project-setup/#_databaseseedprovider[Java Reference -> Extending Neo4j -> Setting up a plugin project].

=== Neo4j configuration settings with changed default values

[options=header, cols="3m,1,1"]
|===
| Name
| Old default value
| New default value

|`dbms.cypher.infer_schema_parts`
|`OFF`
|`MOST_SELECTIVE_LABEL`

|`db.logs.query.annotation_data_format`
|`CYPHER`
|`JSON`

|`dbms.databases.seed_from_uri_providers`
|`S3SeedProvider,CloudSeedProvider`
|`CloudSeedProvider`

|`server.metrics.csv.rotation.compression`
|`NONE`
|`ZIP`

|`server.panic.shutdown_on_panic`
|`false`
|`true`

|===

=== Changed valid values of configuration settings

> The `server.logs.config` is changed to depend on the provided path for the _conf_ directory instead of having the _conf_ directory encoded in the default values.

Author: ```suggestion
The `server.logs.config` and the `server.logs.user.config` are changed to depend on the provided path for the _conf_ directory instead of having the _conf_ directory encoded in the default values.
```

[options=header, cols="2m,3,3"]
|===
| Name
| Old valid value
| New valid value

|`server.logs.config`
|A path. If relative, it is resolved from `server.directories.neo4j_home`.

> |A path. If relative, it is resolved from `server.directories.configuration`.

Author: ```suggestion
|A path. If relative, it is resolved from `server.directories.configuration`.

|`server.logs.user.config`
|A path. If relative, it is resolved from server.directories.neo4j_home.
|A path. If relative, it is resolved from server.directories.configuration.
```

====
[options=header, cols="3m,1,1,3"]
|===
| Name
| Community Edition
| Enterprise Edition
| Comment

| link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_quarantineDatabase[`dbms.quarantineDatabase()`] label:admin-only[]
|
| {check-mark}
| Replaced by xref:procedures.adoc#procedure_dbms_unquarantineDatabase[`dbms.unquarantineDatabase()`]
|===
====

=== Java API

In 2025.01, the deprecated public API is removed.

The `SeedProvider` is removed and replaced by the `DatabaseSeedProvider`.
For details, refer to the link:{neo4j-docs-base-uri}/java-reference/current/extending-neo4j/project-setup/#_databaseseedprovider[Java Reference -> Extending Neo4j -> Setting up a plugin project].

=== Neo4j configuration settings with changed default values

[options=header, cols="3m,1,1"]
|===
| Name
| Old default value
| New default value

|`dbms.cypher.infer_schema_parts`
|`OFF`
|`MOST_SELECTIVE_LABEL`

|`db.logs.query.annotation_data_format`
|`CYPHER`
|`JSON`

|`dbms.databases.seed_from_uri_providers`
|`S3SeedProvider,CloudSeedProvider`
|`CloudSeedProvider`

|`server.metrics.csv.rotation.compression`
|`NONE`
|`ZIP`

|`server.panic.shutdown_on_panic`
|`false`
|`true`

> 

Author: ```suggestion

|`server.logs.config`
|conf/server-logs.xml 
|server-logs.xml

|`server.logs.user.config`
|conf/user-logs.xml
|user-logs.xml
```

:description: Page contains lists of procedures, configuration settings, and metrics removed or deprecated in Neo4j 2025. Also, you can find information on changed defaults and new functionality of neo4j-admin commands.

//Check Mark
:check-mark: icon:check[]

[[removals-deprecations-2025]]
= Neo4j 2025.xx

== Deprecations

> 

Author: ```suggestion

This section lists all of the features that have been deprecated in Neo4j server of 2025 series.
If there are any replacements for deprecated features, they are also indicated.

For deprecations in Cypher language, go to the link:https://neo4j.com/docs/cypher-manual/current/deprecations-additions-removals-compatibility/[Cypher manual -> Deprecations, additions, and compatibility].

```

| xref:configuration/configuration-settings.adoc#config_dbms.security.oidc.-provider-.client_id[`dbms.security.oidc.<provider>.client_id`]
|
| {check-mark}
| label:dynamic[] label:deprecated[]
|===
====

[role=label--enterprise]
=== Metrics

.See all deprecated metrics
[%collapsible]
====
[options=header, cols="3m,3"]
|===
| Name
| Comment

| xref:monitoring/metrics/reference.adoc#db-data-metrics[Database data metrics]
| label:deprecated[Deprecated in 5.15]
|===
====

=== Neo4j-admin tool

.See all deprecated commands
[%collapsible]
====
[options=header, cols="3m,1,1,3"]
|===
| Name
| Community Edition
| Enterprise Edition
| Comment

| link:{neo4j-docs-base-uri}/operations-manual/5/backup-restore/aggregate[`neo4-admin database aggregate-backup`]
|
| {check-mark}
| label:deprecated[Deprecated in 2025.01] +
Replaced by xref:backup-restore/aggregate.adoc[`neo4j-admin backup aggregate`]
|===
====

=== Seed providers

`S3SeedProvider` is deprecated since Neo4j 5.26.
For more information, refer to the xref:clustering/databases.adoc#s3-seed-provider[Seed from URI] documentation on the page xref:clustering/databases.adoc[Managing databases in a cluster].

== Breaking changes

> === Built-in Neo4j procedures

Author: ```suggestion
[role=label--enterprise]
=== Clustering

In 2025.01, discovery service v1 is removed.
Therefore, transitioning from v1 to v2 must be completed before upgrading to Neo4j 2025.01.

For more details, refer to the link:https://neo4j.com/docs/operations-manual/5/clustering/setup/discovery/#clustering-discovery-v1-to-v2[Operations Manual → Moving from discovery service v1 to v2].

=== Built-in Neo4j procedures
```

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.block_cache_size[`server.memory.off_heap.block_cache_size`]
|

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.max_cacheable_block_size[`server.memory.off_heap.max_cacheable_block_size`]
|

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.transaction_max_size[`server.memory.off_heap.transaction_max_size`]
|

|===
====

==== Renamed configuration settings

In Neo4j 2025.01, two configuration settings related to the discovery service v2 were renamed.
You can use still them when migrating from Neo4j 5.26 to the 2025.01 version.
However, it is recommended to update your _neo4j.conf_ file to use the new settings.

[role=label--enterprise]
[options=header, cols="3,3"]
|===
| Old name
| New name

|`dbms.cluster.discovery.v2.endpoints` 
|`dbms.cluster.endpoints`

|`dbms.kubernetes.discovery.v2.service_port_name`
|`dbms.kubernetes.discovery.service_port_name`

|===

==== Changes to default values of configuration settings

> 

Author: ```suggestion

These changes will only affect new installations or upgrades that replace their configuration files.

```

|===
====

[role=label--enterprise]
==== Renamed metric 

In Neo4j 2025.01, the `<prefix>.store.size.total` is renamed to `<prefix>.store.size.full` to be in line with the Prometeus naming requirements.

See the table xref:monitoring/metrics/reference.adoc#db-store-size-metrics[Database store size metrics] for more details.

=== SSL framework

The CBC-based ciphers that are no longer supported from Neo4j 2025.01::

* TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
* TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
* TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
* TLS_DHE_RSA_WITH_AES_128_CBC_SHA256
+
For more details, refer to the page on link:{neo4j-docs-base-uri}/operations-manual/5/security/ssl-framework/#ssl-other-configs[SSL framework] under the Security section.

SSL-related configuration setting::

Starting from 2025.01, the default value of `dbms.ssl.policy.*.verify_hostname` is changed from `false` to `true`.
See link:{neo4j-docs-base-uri}/operations-manual/5/security/ssl-framework/#ssl-configuration[the table on the page _SSL framework_] for more details.

=== Ports

Port `5000` is no longer used from 2025.01 onwards.

> 

Author: ```suggestion
For more details, refer to the xref:configuration/ports.adoc[Configuration -> Ports].
```

| xref:configuration/configuration-settings.adoc#config_dbms.security.oidc.-provider-.client_id[`dbms.security.oidc.<provider>.client_id`]
|
| {check-mark}
| label:dynamic[] label:deprecated[]
|===
====

[role=label--enterprise]
=== Metrics

.See all deprecated metrics
[%collapsible]
====
[options=header, cols="3m,3"]
|===
| Name
| Comment

| xref:monitoring/metrics/reference.adoc#db-data-metrics[Database data metrics]
| label:deprecated[Deprecated in 5.15]
|===
====

=== Neo4j-admin tool

.See all deprecated commands
[%collapsible]
====
[options=header, cols="3m,1,1,3"]
|===
| Name
| Community Edition
| Enterprise Edition
| Comment

| link:{neo4j-docs-base-uri}/operations-manual/5/backup-restore/aggregate[`neo4-admin database aggregate-backup`]
|
| {check-mark}
| label:deprecated[Deprecated in 2025.01] +
Replaced by xref:backup-restore/aggregate.adoc[`neo4j-admin backup aggregate`]
|===
====

=== Seed providers

`S3SeedProvider` is deprecated since Neo4j 5.26.
For more information, refer to the xref:clustering/databases.adoc#s3-seed-provider[Seed from URI] documentation on the page xref:clustering/databases.adoc[Managing databases in a cluster].

> == Breaking changes

Author: @renetapopova, do we need to say something about logging here?

Reviewer: Yeah, that's a good point.
You can add something like:
```
Starting from Neo4j 2025.01, the default _debug.log_ format has been changed from Text to JSON.
It is highly recommended to keep it enabled and not to alter it. +
If you require the _debug.log_ messages in a different format, create an additional Appender since the Neo4j Support team uses the default one.
For more information, see xref:monitoring/logging.adoc##_default_logging_configuration[Default logging configuration].
```

:description: Page contains lists of procedures, configuration settings, and metrics removed or deprecated in Neo4j 2025. Also, you can find information on changed defaults and new functionality of neo4j-admin commands.

//Check Mark
:check-mark: icon:check[]

[[removals-deprecations-2025]]
= Changes, deprecations, and removals in Neo4j 2025

== Breaking changes in Neo4j 2025.01

The section covers breaking changes since the Neo4j 5.26 LTS release.
For guidelines on how to migrate from previous Neo4j versions to 2025.01, see the link:https://neo4j.com/docs/upgrade-migration-guide/current/version-2025[Upgrade and Migration Guide].

[role=label--enterprise]
=== Clustering

Discovery service::
In 2025.01, discovery service v1 is removed.
Therefore, transitioning from v1 to v2 must be completed before upgrading to Neo4j 2025.01.
+
For more details, refer to the link:https://neo4j.com/docs/operations-manual/5/clustering/setup/discovery/#clustering-discovery-v1-to-v2[Operations Manual -> Moving from discovery service v1 to v2].

Configuration settings::

> . In 2025.01, the settings `server.discovery.advertised_address` and `server.discovery.listen_address` are removed.

Reviewer: I think these should be a bullet list, not ordered.

> . In 2025.01, the settings `server.discovery.advertised_address` and `server.discovery.listen_address` are removed.

Reviewer: ```suggestion
* In 2025.01, the settings `server.discovery.advertised_address` and `server.discovery.listen_address` are removed.
```

+
To list the discovery endpoints (e.g., for providing them to the discovery resolver), use the value from the `server.cluster.advertised_address` setting.
+

> . Two configuration settings related to the discovery service v2 are renamed.

Reviewer: ```suggestion
* Two configuration settings related to the discovery service v2 are renamed.
```

You can still use them when migrating from Neo4j 5.26 to the 2025.01 version.
However, it is recommended to update your _neo4j.conf_ file to use the new settings.
+
[options=header, cols="3,3"]
|===
| Old name
| New name

|`dbms.cluster.discovery.v2.endpoints` 
|`dbms.cluster.endpoints`

|`dbms.kubernetes.discovery.v2.service_port_name`
|`dbms.kubernetes.discovery.service_port_name`
|===

Ports::
Port `5000` is no longer used from 2025.01 onwards.
Use port `6000` for the internal traffic.
+
For more details, refer to the xref:configuration/ports.adoc[Configuration -> Ports].

=== Built-in procedures

> .See procedures removed in Neo4j 2025.01 without replacement.

Reviewer: ```suggestion
.See removed procedures without replacement.
```

Author: Honestly, I don't see much harm in mentioning the exact Neo4j version in the table headings

Reviewer: I am more for consistency in this case. So, either have it everywhere or nowhere. On the other hand, having it everywhere might look a bit overloaded, hence my suggestion to remove it from all table titles.

[%collapsible]
====
[options=header,cols="3m,1,1"]
|===
| Name
| Community Edition
| Enterprise Edition

| link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_cluster_movetonextdiscoveryversion[`dbms.cluster.moveToNextDiscoveryVersion()`]
|
| {check-mark}

| link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_cluster_showparalleldiscoverystate[`dbms.cluster.showParallelDiscoveryState()`]
|
| {check-mark}

| link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_cluster_switchdiscoveryserviceversion[`dbms.cluster.switchDiscoveryServiceVersion()`]
|
| {check-mark}

|link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_setDatabaseAllocator[`dbms.setDatabaseAllocator()`]
|
|{check-mark}

|===
====

=== Neo4j configuration settings

==== Removed configuration settings

> .See all removed configuration settings in Neo4j 2025.01

Reviewer: ```suggestion
.See removed configuration settings
```

Reviewer: I am trying to make them look consistent.

|Removed without replacement. +
Instead use the value of xref:configuration/configuration-settings.adoc#config_server.cluster.listen_address[`server.cluster.listen_address`].

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.groups[server.groups] label:enterprise[Enterprise Edition]
|Replaced by xref:configuration/configuration-settings.adoc#config_server.tags[`initial.server.tags`].

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.block_cache_size[server.memory.off_heap.block_cache_size]
|Removed without replacement.

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.max_cacheable_block_size[server.memory.off_heap.max_cacheable_block_size]
|Removed without replacement.

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.transaction_max_size[server.memory.off_heap.transaction_max_size]
|Removed without replacement.

|===
====

==== Changes to default values of configuration settings

> These changes affect only new installations or upgrades that replace existing configuration files.

Reviewer: I understand it as the upgrades can replace config files. Maybe this is what you want to say?
```suggestion
These changes affect only new installations or upgrades that involve replacing existing configuration files.
```

|xref:configuration/configuration-settings.adoc#config_dbms.cypher.infer_schema_parts[dbms.cypher.infer_schema_parts]
|`OFF`
|`MOST_SELECTIVE_LABEL`

|xref:configuration/configuration-settings.adoc#config_db.logs.query.annotation_data_format[db.logs.query.annotation_data_format] label:dynamic[]
|`CYPHER`
|`JSON`

|xref:configuration/configuration-settings.adoc#config_dbms.databases.seed_from_uri_providers[dbms.databases.seed_from_uri_providers] label:enterprise[Enterprise Edition]
|`S3SeedProvider,CloudSeedProvider`
|`CloudSeedProvider`

|xref:configuration/configuration-settings.adoc#config_server.metrics.csv.rotation.compression[server.metrics.csv.rotation.compression] label:enterprise[Enterprise Edition]
|`NONE`
|`ZIP`

|xref:configuration/configuration-settings.adoc#config_server.panic.shutdown_on_panic[server.panic.shutdown_on_panic] label:enterprise[Enterprise Edition]
|`false`
|`true`

|xref:configuration/configuration-settings.adoc#config_server.logs.config[server.logs.config]
|`conf/server-logs.xml`
|`server-logs.xml`

|xref:configuration/configuration-settings.adoc#config_server.logs.user.config[server.logs.user.config]
|`conf/user-logs.xml`
|`user-logs.xml`
|===
====

> ==== Changes to valid values of configuration settings in 2025.01

Reviewer: ```suggestion
==== Changes to valid values of configuration settings
```

The `server.logs.config` and the `server.logs.user.config` are changed to depend on the provided path for the _conf_ directory instead of having the _conf_ directory encoded in the default values.

[options=header, cols="2m,3,3"]
|===
| Name
| Old valid value
| New valid value

|`server.logs.config`
|A path. If relative, it is resolved from `server.directories.neo4j_home`.
|A path. If relative, it is resolved from `server.directories.configuration`.

|`server.logs.user.config`
|A path. If relative, it is resolved from `server.directories.neo4j_home`.
|A path. If relative, it is resolved from `server.directories.configuration`.
|===

[role=label--enterprise]
=== Neo4j metrics

[role=label--enterprise]
==== Removed metrics

> .See all removed metrics in Neo4j 2025.01

Reviewer: ```suggestion
.See all removed metrics
```

In Neo4j 2025.01, the `<prefix>.store.size.total` is renamed to `<prefix>.store.size.full` to be in line with the Prometeus naming requirements.

See the table xref:monitoring/metrics/reference.adoc#db-store-size-metrics[Database store size metrics] for more details.

=== SSL framework

The CBC-based ciphers that are no longer supported from Neo4j 2025.01::

* TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
* TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
* TLS_DHE_RSA_WITH_AES_256_CBC_SHA256
* TLS_DHE_RSA_WITH_AES_128_CBC_SHA256
+
For more details, refer to the page on link:{neo4j-docs-base-uri}/operations-manual/5/security/ssl-framework/#ssl-other-configs[SSL framework] under the Security section.

SSL-related configuration setting::

Starting from 2025.01, the default value of `dbms.ssl.policy.*.verify_hostname` is changed from `false` to `true`.
See link:{neo4j-docs-base-uri}/operations-manual/5/security/ssl-framework/#ssl-configuration[the table on the page _SSL framework_] for more details.

=== Changes to Java API

// For all details about Neo4j Java API, refer to the link:https://neo4j.com/docs/java-reference/2025/javadocs[Neo4j Java API Documentaion].

==== Removals

> . Discovery and clustering

Reviewer: ```suggestion
* Discovery and clustering
```

Reviewer: These shouldn't be ordered unless the order is important.

Classes::
com.neo4j.causalclustering.discovery.parallel.ParallelDiscoveryMode
com.neo4j.causalclustering.discovery.resolve.RemotesResolver.Type

Methods::
ParallelDiscoveryMode::valueOf(java.lang.String)
ParallelDiscoveryMode::values()
RemotesResolver.Type::valueOf(java.lang.String)
RemotesResolver.Type::values()
RemotesResolver::init(com.neo4j.causalclustering.discovery.resolve.RemotesResolver.Type, org.neo4j.graphdb.config.Configuration, org.neo4j.logging.LogProvider)

APIs::
com.neo4j.configuration.ClusterSettings::catchup_connect_randomly_to_server_group
com.neo4j.configuration.ClusterSettings::raft_leader_transfer_priority_group
com.neo4j.configuration.ClusterAddressSettings::discovery_advertised_address
com.neo4j.configuration.DiscoverySettings::discovery_endpoints
com.neo4j.configuration.DiscoverySettings::discovery_listen_address
com.neo4j.configuration.DiscoverySettings::discovery_log_level
com.neo4j.configuration.DiscoverySettings::discovery_type
com.neo4j.configuration.DiscoverySettings::discovery_version +
ClusterSettings::DEFAULT_DISCOVERY_PORT +
ClusterSettings::DEFAULT_RAFT_PORT

> . Seeding and initialization

Reviewer: ```suggestion
* Seeding and initialization
```

Classes::
com.neo4j.dbms.seeding.SeedDownloadStreamWrapper +
com.neo4j.dbms.seeding.SeedProvider +
com.neo4j.dbms.seeding.SeedProviderDependencies

Methods::
SeedDownloadStreamWrapper::dispose() +
SeedDownloadStreamWrapper::getInputStream() +
SeedProvider::download(java.lang.String, java.io.InputStream, java.nio.file.Path) +
SeedProvider::download(java.lang.String, java.util.Optional<java.lang.String>, java.util.Optional<java.lang.String>, java.nio.file.Path) +
SeedProvider::downloadStream(java.lang.String, java.util.Optional<java.lang.String>, java.util.Optional<java.lang.String>, java.nio.file.Path) +
SeedProvider::getDependencies() +
SeedProvider::getLog() +
SeedProvider::getName() +
SeedProvider::getProviderName() +
SeedProvider::inject(com.neo4j.dbms.seeding.SeedProviderDependencies) +
SeedProvider::inject(org.neo4j.logging.Log) +
SeedProvider::matches(java.lang.String) +
SeedProviderDependencies::<T extends java.lang.Object>resolveDependency(java.lang.Class<T>)

APIs::
com.neo4j.configuration.EnterpriseEditionSettings::initial_database_allocator

> . Transaction and memory management

Reviewer: ```suggestion
* Transaction and memory management
```

Classes::
org.neo4j.configuration.GraphDatabaseSettings.TransactionStateMemoryAllocation

Methods::
TransactionStateMemoryAllocation::valueOf(java.lang.String) +
TransactionStateMemoryAllocation::values()

APIs::
org.neo4j.configuration.GraphDatabaseSettings::tx_state_max_off_heap_memory +
org.neo4j.configuration.GraphDatabaseSettings::tx_state_memory_allocation +
org.neo4j.configuration.GraphDatabaseSettings::tx_state_off_heap_block_cache_size +
org.neo4j.configuration.GraphDatabaseSettings::tx_state_off_heap_max_cacheable_block_size

> . Logging and monitoring

Reviewer: ```suggestion
* Logging and monitoring
```

APIs::
org.neo4j.configuration.GraphDatabaseSettings::log_queries_annotation_data_as_json +
com.neo4j.configuration.DiscoverySettings::discovery_log_level

> . General configuration

Reviewer: ```suggestion
* General configuration
```

APIs::
com.neo4j.configuration.EnterpriseEditionSettings::server_groups
com.neo4j.configuration.EnterpriseEditionSettings::server_max_number_of_databases
com.neo4j.configuration.KubernetesSettings::kubernetes_service_port_name

> . Constants

Reviewer: ```suggestion
* Constants
```

:description: Page contains lists of procedures, configuration settings, and metrics removed or deprecated in Neo4j 2025. Also, you can find information on changed defaults and new functionality of neo4j-admin commands.

//Check Mark
:check-mark: icon:check[]

[[removals-deprecations-2025]]

> = Neo4j 2025.xx

Author: I'm not sure about this suggestion. If we change the title to _Changes, deprecations, and removals in Neo4j 2025_, we'll have two same titles in the row at the top

<img width="661" alt="Screenshot 2025-01-22 at 16 29 12" src="https://github.com/user-attachments/assets/9eb0572e-7a0c-4448-8f2a-4c99faa5b347" />

> = Neo4j 2025.xx

Author: If we decide to change the page title, then we need to change the section's title too.

> = Neo4j 2025.xx

Reviewer: Ah, I see what you mean. Hmm, let's think a bit.

> = Neo4j 2025.xx

Author: no probs, we can do that

|Removed without replacement. +
Instead use the value of xref:configuration/configuration-settings.adoc#config_server.cluster.listen_address[`server.cluster.listen_address`].

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.groups[server.groups] label:enterprise[Enterprise Edition]
|Replaced by xref:configuration/configuration-settings.adoc#config_server.tags[`initial.server.tags`].

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.block_cache_size[server.memory.off_heap.block_cache_size]
|Removed without replacement.

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.max_cacheable_block_size[server.memory.off_heap.max_cacheable_block_size]
|Removed without replacement.

|link:{neo4j-docs-base-uri}/operations-manual/5/configuration/configuration-settings#config_server.memory.off_heap.transaction_max_size[server.memory.off_heap.transaction_max_size]
|Removed without replacement.

|===
====

==== Changes to default values of configuration settings

> These changes will only affect new installations or upgrades that replace their configuration files.

Author: I took this sentence from the release notes, as David Pond asked for it.

| xref:configuration/configuration-settings.adoc#config_dbms.security.oidc.-provider-.auth_params[`dbms.security.oidc.<provider>.auth_params`]
|
| {check-mark}
| label:dynamic[] label:deprecated[]

| xref:configuration/configuration-settings.adoc#config_dbms.security.oidc.-provider-.client_id[`dbms.security.oidc.<provider>.client_id`]
|
| {check-mark}
| label:dynamic[] label:deprecated[]
|===
====

[role=label--enterprise]
=== Metrics

.See all deprecated metrics
[%collapsible]
====
[options=header, cols="3m,3"]
|===
| Name
| Comment

| xref:monitoring/metrics/reference.adoc#db-data-metrics[Database data metrics]
| label:deprecated[Deprecated in 5.15]
|===
====

=== Neo4j-admin tool

.See all deprecated commands
[%collapsible]
====
[options=header, cols="3m,1,1,3"]
|===
| Name
| Community Edition
| Enterprise Edition
| Comment

| link:{neo4j-docs-base-uri}/operations-manual/5/backup-restore/aggregate[`neo4-admin database aggregate-backup`]
|
| {check-mark}
| label:deprecated[Deprecated in 2025.01] +
Replaced by xref:backup-restore/aggregate.adoc[`neo4j-admin backup aggregate`]
|===
====

=== Seed providers

`S3SeedProvider` is deprecated since Neo4j 5.26.

> For more information, refer to the xref:clustering/databases.adoc#s3-seed-provider[Seed from URI] documentation on the page xref:clustering/databases.adoc[Managing databases in a cluster].

Reviewer: I think, because this is a direct link to the place where they can find more information, we don't need the second parent link.

> :description: Page contains lists of procedures, configuration settings, and metrics removed or deprecated in Neo4j 2025. Also, you can find information on changed defaults and new functionality of neo4j-admin commands.

Author: ```suggestion
:description: Page lists breaking changes in Neo4j 2025.01 and deprecations in Neo4j 2025 series. Here you can find changes made to procedures, configurations settings, metrics, and `neo4j-admin` commands.
```

//Check Mark
:check-mark: icon:check[]

[[removals-deprecations-2025]]
= Changes, deprecations, and removals in Neo4j 2025.x

== Breaking changes in Neo4j 2025.01

The section covers breaking changes since the Neo4j 5.26 LTS release.
For guidelines on how to migrate from previous Neo4j versions to 2025.01, see the link:https://neo4j.com/docs/upgrade-migration-guide/current/version-2025[Upgrade and Migration Guide].

[role=label--enterprise]
=== Clustering

Discovery service::
In 2025.01, discovery service v1 is removed.

> Therefore, transitioning from v1 to v2 must be completed before upgrading to Neo4j 2025.01.

Reviewer 3: `migrating from` instead of transforming right?

|`conf/server-logs.xml`
|`server-logs.xml`

|xref:configuration/configuration-settings.adoc#config_server.logs.user.config[server.logs.user.config]
|`conf/user-logs.xml`
|`user-logs.xml`
|===
====

==== Changes to valid values of configuration settings

The `server.logs.config` and the `server.logs.user.config` are changed to depend on the provided path for the _conf_ directory instead of having the _conf_ directory encoded in the default values.

.Updated configuration settings
[options=header, cols="2m,3,3"]
|===
| Name
| Old valid value
| New valid value

|`server.logs.config`
|A path. If relative, it is resolved from `server.directories.neo4j_home`.
|A path. If relative, it is resolved from `server.directories.configuration`.

|`server.logs.user.config`
|A path. If relative, it is resolved from `server.directories.neo4j_home`.
|A path. If relative, it is resolved from `server.directories.configuration`.
|===

[role=label--enterprise]
=== Neo4j metrics

[role=label--enterprise]
==== Removed metrics

.See removed metrics in Neo4j 2025.01.
[%collapsible]
====
[options="header", cols="1,1"]
|===
|Name|Description

> 2+|**link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#raft-core-metrics[Raft core metrics] - replaced accordingly by the link:{neo4j-docs-base-uri}/operations-manual/5/monitoring/metrics/reference/#raft-metrics[Raft metrics]**

Reviewer: We don't need descriptions on this page. I think having a simple table with the old metric(link) and the replacement (link) would make it easier to grasp the change.

Author: We can improve it later. BTW, metrics don't have their own links. They're grouped in tables. I tried different scenarios here.  We can leave here only metrics classes, but that is not very informative

Reviewer: Sure. Sounds like a project 😄

|
| {check-mark}
| label:deprecated[Deprecated in 5.6] +
Replaced by: xref:procedures.adoc#procedure_dbms_cluster_secondaryreplicationdisable[`dbms.cluster.secondaryReplicationDisable()`].
|===
====

=== Configuration settings

.See all deprecated configuration settings
[%collapsible]
====
[options=header, cols="3m,1,1,2"]
|===
| Name
| Community Edition
| Enterprise Edition
| Comment

| xref:configuration/configuration-settings.adoc#config_server.db.query_cache_size[`server.db.query_cache_size`]
| {check-mark}
| {check-mark}
| label:deprecated[Deprecated in 5.7]

| xref:configuration/configuration-settings.adoc#config_dbms.security.oidc.-provider-.auth_params[`dbms.security.oidc.<provider>.auth_params`]
|
| {check-mark}
| label:dynamic[] label:deprecated[]

| xref:configuration/configuration-settings.adoc#config_dbms.security.oidc.-provider-.client_id[`dbms.security.oidc.<provider>.client_id`]
|
| {check-mark}
| label:dynamic[] label:deprecated[]
|===
====

[role=label--enterprise]
=== Metrics

.See all deprecated metrics
[%collapsible]
====
[options=header, cols="3m,3"]
|===
| Name
| Comment

> | xref:monitoring/metrics/reference.adoc#db-data-metrics[Database data metrics]

Reviewer: I think we should list all deprecated database data metrics with their replacements where available.

Reviewer: Some of them have kind of replacements. Let me find those... For example:

<prefix>.ids_in_use.relationship => "If you want to have a rough metric of how big your graph is, use <prefix>.neo4j.count.relationship instead."
<prefix>.ids_in_use.node => "If you want to have a rough metric of how big your graph is, use <prefix>.neo4j.count.node instead."

Author: oh, yeah. Right! This information in in their descriptions. I missed it. Thank you

|WaitResponseState::valueOf(java.lang.String) +
WaitResponseState::values()

|APIs
|ClusterSettings::DEFAULT_CLUSTER_STATE_DIRECTORY_NAME +
ClusterSettings::DEFAULT_DISCOVERY_PORT +
ClusterSettings::DEFAULT_RAFT_PORT +
ClusterSettings::DEFAULT_TRANSACTION_PORT +
ClusterBaseSettings::DEFAULT_DISCOVERY_PORT +
RaftSettings::DEFAULT_CLUSTER_STATE_DIRECTORY_NAME +
RaftSettings::DEFAULT_RAFT_PORT
|===
====

The `SeedProvider` is removed and replaced by the `DatabaseSeedProvider`.
For details, refer to the link:{neo4j-docs-base-uri}/java-reference/current/extending-neo4j/project-setup/#_databaseseedprovider[Java Reference -> Extending Neo4j -> Setting up a plugin project].

== Deprecations in Neo4j 2025.x

This section lists all the Neo4j Server features deprecated in the 2025 series, including replacements where applicable.

For deprecations in Cypher language, see link:https://neo4j.com/docs/cypher-manual/current/deprecations-additions-removals-compatibility/[Cypher manual -> Deprecations, additions, and compatibility].

=== Procedures

.See all deprecated procedures
[%collapsible]
====
[options=header, cols="3m,1,1,3"]
|===
| Name
| Community Edition
| Enterprise Edition
| Comment

> | link:{neo4j-docs-base-uri}/operations-manual/5/procedures/#procedure_dbms_quarantineDatabase[`dbms.quarantineDatabase()`] label:admin-only[]

Author: ```suggestion
| xref:procedures.adoc#procedure_dbms_quarantineDatabase[`dbms.quarantineDatabase()`] label:admin-only[]
```
